### PR TITLE
Add fuzzers for loading multiple focus functions

### PR DESCRIPTION
**Pull request by Navidem**

### CODE REVIEW

**Review by Navidem — COMMENTED** [review]
fix indices

**Review by oliverchang — APPROVED** [review]
LGTM. Note that Centipede allows multiple focus functions (https://github.com/google/centipede/blob/c9f0ea0458d081addc39abfbf89ac3ebd5b804cb/environment.cc#L204), which may simplify this.

**Comment by Navidem** [comment]
> LGTM. Note that Centipede allows multiple focus functions (https://github.com/google/centipede/blob/c9f0ea0458d081addc39abfbf89ac3ebd5b804cb/environment.cc#L204), which may simplify this.

Yes! I am aware of that.

**Comment by Navidem** [comment]
CI failures are not reproducible locally. For example CI fails with:
`make: *** [docker/generated.mk:176501: .libfuzzer_focus_idx0-poppler_pdf_fuzzer-builder] Error 1`

while `make test-run-libfuzzer_focus_idx0-poppler_pdf_fuzzer` and `make debug-libfuzzer_focus_idx0-poppler_pdf_fuzzer` finish without any error locally.

**Review by jonathanmetzman — APPROVED** [review]
LGTM

File: .github/workflows/fuzzers.yml [hunk 1]
```diff
@@ -96,6 +96,15 @@ jobs:
           - afl_virginmap
           - afl_maxmap
           - introspector_driven_focus
+          - libfuzzer_focus_idx0
```
Comment (jonathanmetzman) [inline .github/workflows/fuzzers.yml:99]: meh, don't bother putting all of these in our CI. This is too much

**Comment by jonathanmetzman** [comment]
> make test-run-libfuzzer_focus_idx0-poppler_pdf_fuzzer

What's the failure? Github is not loading the logs for me.

File: .github/workflows/fuzzers.yml (hunk 1 above)
Comment (jonathanmetzman) [reply .github/workflows/fuzzers.yml:99]: Right. I wonder if this will fail for future people adding fuzzers. I guess if it does, then it is not worth it.

**Comment by Navidem** [comment]
> > make test-run-libfuzzer_focus_idx0-poppler_pdf_fuzzer
> 
> What's the failure? Github is not loading the logs for me.

something like this:

```
2022-08-24T04:11:39.9221114Z #17 29.98 -- no fontconfig or win32 specific code
2022-08-24T04:11:39.9221539Z #17 29.98 -- Found Freetype: /work/lib (Required is at least version "2.8") 
2022-08-24T04:11:39.9222004Z #17 29.99 -- Could NOT find JPEG (missing: JPEG_LIBRARY JPEG_INCLUDE_DIR) 
2022-08-24T04:11:39.9222488Z #17 30.00 -- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.8") 
2022-08-24T04:11:39.9222998Z #17 30.00 -- Could NOT find PNG (missing: PNG_LIBRARY PNG_PNG_INCLUDE_DIR) 
2022-08-24T04:11:39.9223460Z #17 30.01 -- Could NOT find TIFF (missing: TIFF_LIBRARY TIFF_INCLUDE_DIR) 
2022-08-24T04:11:39.9223920Z #17 30.01 -- Package Qt6Core or Qt6Gui or Qt6Widgets or Qt6Test not found
2022-08-24T04:11:39.9224435Z #17 30.05 -- Could NOT find Boost (missing: Boost_INCLUDE_DIR) (Required is at least version "1.58.0")
2022-08-24T04:11:40.0100941Z #17 30.05 CMake Error at CMakeLists.txt:202 (message):
2022-08-24T04:11:40.0127874Z #17 30.05   -- Boost recommended for Splash.  Use ENABLE_BOOST=OFF to skip.
2022-08-24T04:11:40.0128299Z #17 30.05 
2022-08-24T04:11:40.0128511Z #17 30.05 
2022-08-24T04:11:40.0128930Z #17 30.05 -- Configuring incomplete, errors occurred!
2022-08-24T04:11:40.0130228Z #17 30.05 See also "/work/poppler/CMakeFiles/CMakeOutput.log".
2022-08-24T04:11:40.0130552Z #17 30.06 Traceback (most recent call last):
2022-08-24T04:11:40.0130839Z #17 30.06   File "<string>", line 1, in <module>
2022-08-24T04:11:40.0131174Z #17 30.06   File "/src/fuzzers/libfuzzer_focus_idx0/fuzzer.py", line 24, in build
2022-08-24T04:11:40.0131493Z #17 30.06     fuzzer.build()
2022-08-24T04:11:40.0132082Z #17 30.06   File "/src/fuzzers/libfuzzer/fuzzer.py", line 35, in build
2022-08-24T04:11:40.0132380Z #17 30.06     utils.build_benchmark()
2022-08-24T04:11:40.0134449Z #17 30.06   File "/src/fuzzers/utils.py", line 82, in build_benchmark
2022-08-24T04:11:40.0135008Z #17 30.06     subprocess.check_call(['/bin/bash', '-ex', build_script], env=env)
2022-08-24T04:11:40.0136752Z #17 30.06   File "/usr/local/lib/python3.8/subprocess.py", line 364, in check_call
2022-08-24T04:11:40.0137117Z #17 30.06     raise CalledProcessError(retcode, cmd)
2022-08-24T04:11:40.0137695Z #17 30.06 subprocess.CalledProcessError: Command '['/bin/bash', '-ex', '/src/build.sh']' returned non-zero exit status 1.
2022-08-24T04:11:40.0141890Z #17 ERROR: executor failed running [/bin/sh -c echo "Run fuzzer_build to build the target" && if [ -z "$debug_builder" ] ; then fuzzer_build; fi]: exit code: 1
```

File: .github/workflows/fuzzers.yml (hunk 1 above)
Comment (Navidem) [reply .github/workflows/fuzzers.yml:99]: Yes! it may cause others see their CI failing too.

Comment (Navidem) [reply .github/workflows/fuzzers.yml:99]: Should I skip registering it and ignore presubmit failure?

Comment (jonathanmetzman) [reply .github/workflows/fuzzers.yml:99]: Nah, leave as is please.

**Comment by jonathanmetzman** [comment]
I get local failure when running: `make test-run-libfuzzer_focus_idx0-libpng-1.2.5`

**Comment by jonathanmetzman** [comment]
```docker run \
--cpus=1 \
--shm-size=2g \
--cap-add SYS_NICE \
--cap-add SYS_PTRACE \
-e FUZZ_OUTSIDE_EXPERIMENT=1 \
-e FORCE_LOCAL=1 \
-e TRIAL_ID=1 \
-e FUZZER=libfuzzer_focus_idx0 \
-e BENCHMARK=libpng-1.2.56 \
-e FUZZ_TARGET=libpng_read_fuzzer \
-e MAX_TOTAL_TIME=20 \
-e SNAPSHOT_PERIOD=10 \
gcr.io/fuzzbench/runners/libfuzzer_focus_idx0/libpng-1.2.56
INFO:root:Doing trial as part of experiment.
INFO:root:Starting trial.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/src/fuzzers/libfuzzer_focus_idx0/fuzzer.py", line 42, in fuzz
    fuzzer.run_fuzzer(input_corpus,
  File "/src/fuzzers/libfuzzer/fuzzer.py", line 61, in run_fuzzer
    os.environ['ASAN_OPTIONS'] += ':symbolize=1'
  File "/usr/local/lib/python3.8/os.py", line 675, in __getitem__
    raise KeyError(key) from None
KeyError: 'ASAN_OPTIONS'
ERROR:root:Executed command: "nice -n 5 python3 -u -c from fuzzers.libfuzzer_focus_idx0 import fuzzer; fuzzer.fuzz('/out/seeds', '/out/corpus', '/out/libpng_read_fuzzer')" returned: 1.
ERROR:root:Fuzz process returned nonzero. Extras: {'traceback': 'Traceback (most recent call last):\n  File "/src/experiment/runner.py", line 223, in run_fuzzer\n    new_process.execute(command,\n  File "/src/common/new_process.py", line 124, in execute\n    raise subprocess.CalledProcessError(retcode, command)\nsubprocess.CalledProcessError: Command \'[\'nice\', \'-n\', \'5\', \'python3\', \'-u\', \'-c\', "from fuzzers.libfuzzer_focus_idx0 import fuzzer; fuzzer.fuzz(\'/out/seeds\', \'/out/corpus\', \'/out/libpng_read_fuzzer\')"]\' returned non-zero exit status 1.\n'}
INFO:root:Doing final sync.
make: *** [docker/generated.mk:235749: test-run-libfuzzer_focus_idx0-libpng-1.2.56] Error```

**Comment by Navidem** [comment]
> I get local failure when running: `make test-run-libfuzzer_focus_idx0-libpng-1.2.56`

Now it passes.

**Comment by Navidem** [comment]
> I get local failure when running: `make test-run-libfuzzer_focus_idx0-libpng-1.2.5`

Now it should pass.

**Comment by Navidem** [comment]
CI all passed, except the build for `bug` which hit timeout.

**Pull request merged**